src/UI/Components/Diagram/halo.ts

```diff
@@ -43,16 +44,29 @@ const createHalo = (
       const isEmbeddedToTHisCell =
         element.get("embeddedTo") === cellView.model.id;
 
+      let didElementChange = false;
+
       if (isEmbedded && isEmbeddedToTHisCell) {
         element.set("embeddedTo", null);
+        didElementChange = true;
       }
       const relations = elementAsService.getRelations();
 
       if (relations) {
-        elementAsService.removeRelation(cellView.model.id as string);
+        didElementChange =
+          didElementChange === true ||
```

Comment:
The or statement still confuses me. Does it mean that if the Element already changed, you don't remove the relation?

the removeRelation method serves two purposes, maybe it would be best to split it?

Reply:
Yes, because if element changed, that means that connection that was made to that element is already removed, there is no way that we have both embedded entity connection and inter-service relation between two shapes

Reply:
I can definitely do that if you have feeling that it's confusing, I just accepted it, as the default behaviour of function on which we are basing is giving that opportunity not to delete and check whether something changed

Reply:
yeah, but if you read the code now, you assume that you'll delete the relation if the element hasn't changed and it has relations.